TTS/tts/models/base_tts.py

```diff
@@ -60,13 +60,22 @@ def init_multispeaker(self, config: Coqpit, data: List = None):
         # init speaker manager
         self.speaker_manager = get_speaker_manager(config, data=data)
```

Comment:
Why do you pass `data` if it is not used at all? 

I also don't like the idea to pass `data`to init a model. It doesn't make sense to me.

Reply:
What do you mean it's not used at all? It's used in the speaker manager to parse the diffrent speakers from the dataset no? https://github.com/coqui-ai/TTS/blob/c25a2184e72b0bc28f4ff8f8886cc01d6f7f5c5b/TTS/tts/utils/speakers.py#L316

Reply:
> I also don't like the idea to pass data to init a model. 

What do you suggest then? Initializing the SpeakerManager before initializing the model, and passing the manager to the model at initialization?

Reply:
I don't think you need the speaker manager in the model. It can live outside of the model and just provide the inputs when needed.

Reply:
you need to parse the data to find the number of speakers right?

Reply:
also you just need to do it at training since at inference you have the speakers.json ready?

If I am not missing something, we can use `on_init_start()` callback to call the `init_multi_speaker` as all the callbacks access the trainer and the model at the same time. Then you can pass the data from the trainer to the init_multi_speaker


makes sense?

Reply:
Sounds good to me

Reply:
Now that I think about it, I don't think your idea would work since __init__ of tacotron would have already been called before on_init_start() is called. So the model will initialize badly for exemple here where __init__ needs to know self.embedded_speaker_dim that is determined by init_multispeaker https://github.com/WeberJulian/TTS-1/blob/4b1c06474c1b016dc95d367456d4106b1cef8a93/TTS/tts/models/tacotron2.py#L36 .
Do you agree with my assessment ?

Reply:
Yeah you are right :(

One easy fix no headache is expecting users to set the right number of speakers manually. How about that?

Reply:
An option is to read the dataset under `get_speaker_manager`and detect the number of speakers.

Reply:
> One easy fix no headache is expecting users to set the right number of speakers manually.

I don't think it's a good solution because depending on the number of datasets and speakers per dataset it can become cumbersome. And it's another point of failure for the user to make a mistake.

Reply:
> An option is to read the dataset under `get_speaker_manager`and detect the number of speakers.

I'm not sure I understand. It's already what I'm doing by passing data to `get_speaker_manager(config, data=data)` and it's called during the __init__ of the model.

Reply:
What I mean is to call load_meta_data in get_speaker_manager as we pass the dataset parameters with config. Then you don't need to pass data but parse it in the function.

Reply:
Oh ok, so it would have to load the dataset twice ? It would log things twice and take twice as long no? But if you think it's better, it's fine by me.

Reply:
load_meta_data logs only a single line. We can set an argument `verbose=False` and set it False by default to prevent it to be printed.